frontend/model/contracts/group.js

```diff
@@ -1286,8 +1276,11 @@ sbp('chelonia/defineContract', {
       process ({ meta }, { state, getters }) {
         const period = getters.periodStampGivenDate(meta.createdDate)
         const current = getters.groupSettings?.distributionDate
-
-        if (current !== period) {
+        const inWaitingPeriod = !current || meta.createdDate < current
```

Comment:
Good catch, but couldn't I just use the new `getters.groupDistributionStarted(meta.createdDate)` implemented by Alex?

Although I see it's not used alone somewhere else in this file, but combined with a check for the number of stored period keys:

          if (getters.groupDistributionStarted(meta.createdDate) || Object.keys(getters.groupPeriodPayments).length > 1)) {

(not sure why the getter isn't enough to ensure we're out of the waiting period)

Reply:
> In a "normal" (not waiting) period, meta.createdDate < current should always be false, since the distribution date is when the current period started, not when the next period will start

That's probably true.

> couldn't I just use the new getters.groupDistributionStarted(meta.createdDate) implemented by Alex?

The two seem to be equivalent. You could do:

```js
const inWaitingPeriod = !getters.groupDistributionStarted(meta.createdDate)
```

That would probably be best from a DRY perspective.

> but combined with a check for the number of stored period keys:

Yeah I'm not sure what that's about. @Silver-IT couldn't you remove that `|| Object.keys(getters.groupPeriodPayments).length > 1` check? Although that might be there for a reason related to adjusting the distribution date that I don't remember.

Yeah, it could be that the distribution date was adjusted via a proposal to suddenly be in the past?

This is the code: https://github.com/okTurtles/group-income/pull/1643/files#diff-5c6306168381a1d6902fce7da1cdf9a6811e2ed4ac74d7d97bfff5145087b626R122

It seems like the proposal only allows for updating the distribution date to a date in future, starting with 1 day in the future.

However, what if by the time the proposal passes we are already ahead of that date? For example, what today is Sept. 1st, and we set the distribution date to Sept 30th, but then create a proposal to make it Sept. 2nd, but by the time the proposal passes it's already Sept 3rd?

Need to consider this.... 🤕